tests/core/test_kontainer_drivers.py

```diff
@@ -97,6 +98,58 @@ def test_enabling_driver_exposes_schema(admin_mc, remove_resource):
     verify_driver_not_in_types(admin_mc.client, kd)
 
 
+@pytest.mark.nonparallel
+def test_create_duplicate_driver_conflict(admin_mc, wait_remove_resource):
+    """ Test if adding a driver with a pre-existing driver's URL
+    returns a conflict error"""
+    kd = admin_mc.client.create_kontainerDriver(
+        createDynamicSchema=True,
+        active=True,
+        url=DRIVER_URL
+    )
+    wait_remove_resource(kd)
+    kd = wait_for_condition('Active', 'True', admin_mc.client, kd, timeout=90)
+
+    try:
+        kd2 = admin_mc.client.create_kontainerDriver(
+            createDynamicSchema=True,
+            active=True,
+            url=DRIVER_URL
+        )
+        wait_remove_resource(kd2)
+    except ApiError as e:
+        assert e.error.status == 409
+        assert "Driver URL already in use:" in e.error.message
```

Comment:
What if the driver successfully creates, will your test fail? It doesn't look like it will.  Try adding a `fail` after `wait_remove_resource` in the try block.